internal/certdb/validation.go

```diff
@@ -0,0 +1,56 @@
+package certdb
+
+import (
+	"crypto/rsa"
+	"crypto/x509"
+	"encoding/pem"
+	"errors"
+)
+
+// ValidateCertificateRequest validates the given CSR string to the following:
+// The string must be a valid PEM string, and should be of type CERTIFICATE REQUEST
+// The PEM string should be able to be parsed into a x509 Certificate Request
+func ValidateCertificateRequest(csrString string) error {
+	block, _ := pem.Decode([]byte(csrString))
+	if block == nil {
+		return errors.New("PEM Certificate Request string not found or malformed")
+	}
+	if block.Type != "CERTIFICATE REQUEST" {
+		return errors.New("given PEM string not a certificate request")
+	}
+	_, err := x509.ParseCertificateRequest(block.Bytes)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// ValidateCertificate validates the given Cert string to the following:
+// The given CSR must pass the validation provided by ValidateCertificateRequest
+// The cert string must be a valid PEM string, and should be of type CERTIFICATE
+// The PEM string should be able to be parsed into a x509 Certificate
+// The given cert and CSR must share the same public key
+func ValidateCertificate(certString string, csrString string) error {
```

Comment:
I can imagine a function that verifies the format of the certificate like done in `ValidateCertificateRequest` being useful on its own.
Should this be two separate functions?

Reply:
I avoided doing that because I didn't want to give the impression that these are helper functions to be used anywhere we need to look at certificates, but there is really no downside to doing that if that's preferred. I've changed it.